src/core/main.c

```diff
@@ -652,107 +652,6 @@ static int config_parse_crash_chvt(
         return 0;
 }
 
-static int config_parse_join_controllers(const char *unit,
-                                         const char *filename,
-                                         unsigned line,
-                                         const char *section,
-                                         unsigned section_line,
-                                         const char *lvalue,
-                                         int ltype,
-                                         const char *rvalue,
-                                         void *data,
-                                         void *userdata) {
-
-        const char *whole_rvalue = rvalue;
-        unsigned n = 0;
-
-        assert(filename);
-        assert(lvalue);
-        assert(rvalue);
-
-        arg_join_controllers = strv_free_free(arg_join_controllers);
```

Comment:
Equally, having to use arg_join_controllers directly instead of through data was a sign we were being a bit weird.  Nice cleanup :).